pkg/kubectl/cmd/delete.go

```diff
@@ -254,7 +267,10 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 			// If there is no reaper for this resources and the user didn't explicitly ask for stop.
 			if kubectl.IsNoSuchReaperError(err) && isDefaultDelete {
 				// No client side reaper found. Let the server do cascading deletion.
-				return cascadingDeleteResource(info, out, shortOutput, mapper)
+				if err := deleteResourceAndWait(info, out, shortOutput, mapper, false, waitForDeletion, timeout); err != nil {
```

Comment:
Added the TODO and also added a link to https://github.com/kubernetes/kubernetes/issues/46659 for more details